### PR TITLE
Feature/main data classes tasks

### PR DESCRIPTION
**Pull request by rodikno**
Main data classes tasks are done

### CODE REVIEW

File: Main_Tasks/Main_Data_Classes/main_data_classes_task.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require_relative "helper_methods"
+
+#method takes natural number and returns sum of its digits
+def sum_of_digits(number)
+  a, b = [], []
+  sum = 0
+  if is_natural_number(number)
+    0.upto(number.size) {
```
Comment (d-konovalov) [inline Main_Tasks/Main_Data_Classes/main_data_classes_task.rb]: number.size returns size in bytes and it's not what we look for actually

File: Main_Tasks/Main_Data_Classes/main_data_classes_task.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      a = number.divmod(10)
```
Comment (d-konovalov) [inline Main_Tasks/Main_Data_Classes/main_data_classes_task.rb]: completely not clear what happened here

File: Main_Tasks/Main_Data_Classes/main_data_classes_task.rb [hunk 3]
```diff
@@ -36,3 +36,17 @@ def number_of_first_matching_chars(string)
   end
   matching_array.length
 end
+
+def even_odd_by_index(array)
```
Comment (d-konovalov) [inline Main_Tasks/Main_Data_Classes/main_data_classes_task.rb:43]: this one doesn't work

File: Main_Tasks/Main_Data_Classes/main_data_classes_task.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  print "Even: #{even}\nOdd: #{odd}"
+end
+
+#returns index of last elem which follows the condition a[0] < elem < a[-1]
+def last_element_in_range_index(array)
+  range = (array.first+1...array.last)
+  indexes_of_elements_in_range = []
+
+  array.each_with_index do |elem, i|
+    indexes_of_elements_in_range.push(i) if range.include?(elem)
```
Comment (d-konovalov) [inline Main_Tasks/Main_Data_Classes/main_data_classes_task.rb]: you shouldn't check whether element in the range but should compare value of the element with first and last elements' values

File: Main_Tasks/Main_Data_Classes/main_data_classes_task.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  end
+  indexes_of_elements_in_range.last
+end
+
+#modifies given array by adding first element to each even number
+def add_first_elem_to_each_even(array)
+  range = (1...array.length-1)
+  array.each_with_index do |elem, index|
+    if range.include?(index) && elem.even?
```
Comment (d-konovalov) [inline Main_Tasks/Main_Data_Classes/main_data_classes_task.rb]: Why we need here this check:

```
range.include?(index)
```
?

File: Main_Tasks/Main_Data_Classes/main_data_classes_task.rb [hunk 6]
```diff
@@ -1,18 +1,15 @@
...
-  a, b = [], []
   sum = 0
   if is_natural_number(number)
-    0.upto(number.size) {
-      a = number.divmod(10)
-      sum += a[1]
-      number = a[0]
-    }
-    sum
+    string_number = number.to_s
```
Comment (rodikno) [inline Main_Tasks/Main_Data_Classes/main_data_classes_task.rb:7]: @d-konovalov now It's working without any wrong `.size` and strange solutions :)

File: Main_Tasks/Main_Data_Classes/main_data_classes_task.rb [hunk 7]
```diff
@@ -54,15 +54,14 @@ def even_odd_by_index(array)
   print "Even: #{even}\nOdd: #{odd}"
 end
 
-#returns index of last elem which follows the condition a[0] < elem < a[-1]
-def last_element_in_range_index(array)
-  range = (array.first+1...array.last)
-  indexes_of_elements_in_range = []
+#returns index of last elem of a given array which follows the condition a[0] < elem < a[-1]
+def last_element_conditional_index(array)
```
Comment (rodikno) [inline Main_Tasks/Main_Data_Classes/main_data_classes_task.rb:58]: @d-konovalov changed the condition, now checking the exact occurrence without using range

Comment (rodikno) [inline Main_Tasks/Main_Data_Classes/main_data_classes_task.rb]: @d-konovalov changed the condition, now checking the exact occurrence without using range

**Pull request merged**